packages/control-hub/src/internal/ControlHub.ts

```diff
@@ -0,0 +1,794 @@
+    responseTimeoutMs: number = 0;
+    type: RevHubType = RevHubType.ControlHub;
+    readonly serialNumber: string;
+    readonly children: ReadonlyArray<RevHub> = [];
+
+    /**
+     * Whether the websocket is currently open.
+     */
+    isConnected = false;
+
+    private webSocketConnection!: WebSocket;
+    private keyGenerator = 0;
+    private currentActiveCommands = new Map<
+        any,
+        (response: any | undefined, error: any | undefined) => void
+    >();
+
+    constructor(serialNumber: string) {
+        this.serialNumber = serialNumber;
+    }
+
+    isParent(): this is ParentRevHub {
+        return this.serialNumber !== undefined;
+    }
+
+    async open(ip: string = "192.168.43.1", port: string = "8081"): Promise<void> {
+        this.webSocketConnection = new WebSocket(`ws://${ip}:${port}`);
+
+        this.webSocketConnection.on("message", (data) => {
+            let rawMessage = JSON.parse(data.toString());
+
+            if (rawMessage.commandKey !== undefined) {
+                let key = rawMessage.commandKey;
+                let callback = this.currentActiveCommands.get(key);
+
+                let response = rawMessage.response
+                    ? JSON.parse(rawMessage.response)
+                    : undefined;
+                let error = rawMessage.error ? JSON.parse(rawMessage.error) : undefined;
+
+                if (callback) {
+                    callback(response, error);
+                }
+            }
+        });
+
+        this.webSocketConnection.on("close", () => {
+            console.log(`Connection was closed`);
+            this.isConnected = false;
+        });
+
+        this.webSocketConnection.on("error", (_: WebSocket, err: Error) => {
+            console.log("Websocket error");
+            console.log(err);
+            this.isConnected = false;
+        });
+
+        return new Promise((resolve, reject) => {
+            this.webSocketConnection.on("open", async () => {
+                this.isConnected = true;
+                console.log("Starting manual control op mode");
+                let apiVersion: { majorVersion: string; minorVersion: string } =
+                    await this.sendCommand("start", {});
+
+                console.log(
+                    `API version is ${apiVersion.majorVersion}.${apiVersion.minorVersion}`,
+                );
+
+                this.moduleAddress = await this.sendCommand("getModuleAddress", {
+                    serialNumber: "Embedded",
+                    moduleAddress: 173,
+                });
+                resolve();
+            });
+        });
+    }
+
+    async isWiFiConnected(): Promise<boolean> {
+        try {
+            let response = await axios.get("http://192.168.43.1:8080/js/rcInfo.json", {
+                timeout: 1000,
+            });
+            if (response.data) {
+                let rcVersion: string = response.data.rcVersion;
+                return semver.satisfies(rcVersion, ">=8.2");
```

Comment:
Use `sdkVersion` instead of `rcVersion` (from https://github.com/FIRST-Tech-Challenge/ftc_sdk/pull/2926)